scripts/ci/gitlab/pipeline/test.yml

```diff
@@ -435,3 +411,32 @@ check-rustdoc:
     - rusty-cachier snapshot create
     - time cargo +nightly doc --locked --workspace --all-features --verbose --no-deps
     - rusty-cachier cache upload
+
+.cargo-check-each-crate_template: &cargo-check-each-crate_template
+  stage:                           test
+  extends:
+    - .docker-env
+    - .test-refs
+  parallel: 2
```

Comment:
FWIW we have 296 crates on Substrate at the moment

```console
$ cargo tree --workspace --depth 0 | awk '{ if (length($1) == 0 || substr($1, 1, 1) == "[") { skip } else { print $1 } }' | wc -l
296
```

Perhaps it's worth it to have even more runners for this, but I think 2 is fine for a start